packages/adaptivecpp/package.py

```diff
@@ -52,21 +52,69 @@ class Adaptivecpp(CMakePackage):
         submodules=True,
     )
 
+    default_compilationflow = "omp-library-only"
+    variant(
+        "compilationflow",
+        default=default_compilationflow,
+        values=(
+            default_compilationflow,
+            "omp-accelerated",
+            "cuda-llvm",
+            "cuda-nvcxx",
+        ),
+        description="Specify the default compilation workflow which this install will use for all translation units. Setting this variant will automatically select other variants as needed. For cuda compilation flows the CUDA architecture should be set with, e.g. 'cuda_arch=80'. The cuda-llvm flow requires that cuda_arch is set.",
+        multi=False,
+    )
+
+    default_cuda_arch = "none"
+    cuda_arch_values = tuple(
+        [default_cuda_arch] + list(CudaPackage.cuda_arch_values)
+    )
+    variant(
+        "cuda_arch",
+        default=default_cuda_arch,
+        values=cuda_arch_values,
+        description="Specify the CUDA architecture to use. Required, i.e. not 'none', for cuda-llvm compilation flow.",
+        multi=False,
+    )
+    conflicts("cuda_arch=none", when="compilationflow=cuda-llvm")
+
     variant(
         "cuda",
         default=False,
         description="Enable CUDA backend for SYCL kernels using llvm+cuda",
     )
+    variant(
+        "cuda",
+        default=True,
+        when="compilationflow=cuda-llvm",
+        description="Enable CUDA backend for SYCL kernels using llvm+cuda",
+    )
+    conflicts("~cuda", when="compilationflow=cuda-llvm")
     variant(
         "nvcxx",
         default=False,
         description="Enable CUDA backend for SYCL kernels using nvcxx",
     )
+    variant(
+        "nvcxx",
+        default=True,
+        when="compilationflow=cuda-nvcxx",
+        description="Enable CUDA backend for SYCL kernels using nvcxx",
+    )
+    conflicts("~nvcxx", when="compilationflow=cuda-nvcxx")
     variant(
         "omp_llvm",
         default=False,
         description="Enable accelerated OMP backend for SYCL kernels using LLVM",
     )
+    variant(
+        "omp_llvm",
+        default=True,
+        when="compilationflow=omp-accelerated",
+        description="Enable accelerated OMP backend for SYCL kernels using LLVM",
+    )
+    conflicts("~omp_llvm", when="compilationflow=omp-accelerated")
```

Comment:
Aren't these made redundant by having `compilationflow`? Why do you need these variants as well?

Reply:
If you do want an install of acpp that has other backends then these are useful to have. It is useful to still be able to do `spack install neso.adaptivecpp +cuda +omp_llvm` or hopefully soon `spack install neso.adaptivecpp +cuda + omp_llvm +generic` and have one install that has multiple backends. 

I am proposing the `compilationflow=foo` as the recomended default for downstream places like neso/neso-tokamak/Reactions.